### DISCUSSION TITLE
Why AES only works when the problem size 65536

### DISCUSSION THREAD

**Discussion by Yzgll** [sarchlab/mgpusim#54] [Q&A] [answered]
Why does it prompt a bad match when I run AES on mgpusim no matter how much I modify the length, the CPU result is inconsistent with the GPU result, and only when the length is the default 65536 it works correctly?

**Comment by syifan** [comment, marked as answer]
I think AES only works with block sizes of 64 (or 256, I do not remember clearly). Can you try with some power of 2? Say 1048576?

**Reply by Yzgll** [reply]
I'm glad to receive your reply. This was a problem I hadn't considered before, but I've successfully resolved it now. MGPUSIM is a great simulator, and I'll study it diligently. Thank you very much!

**Reply by syifan** [reply]
Thank you very much for your kind words! I will close this discussion then.
